src/integrationTest/java/uk/gov/hmcts/reform/cpo/controllers/BaseMvcAuthChecks.java

```diff
@@ -0,0 +1,281 @@
+     */
+    default void assert2xxSuccessfulForHappyPath(MockMvc mockMvc,
+                                                 MockHttpServletRequestBuilder happyPathRequestBuilder,
+                                                 String serviceName) throws Exception {
+        // GIVEN
+        HttpHeaders headers = createHttpHeaders(serviceName); // <-- VALID AUTH & S2S AUTH
+
+        // WHEN
+        mockMvc.perform(happyPathRequestBuilder.headers(headers))
+            // THEN
+            .andExpect(status().is2xxSuccessful());
+    }
+
+    /*
+     * CPO-33: “Implement tests for invalid S2S and Idam token”
+     * AC2: Mandatory parameters missing from the request (IDAM token Missing)
+     */
+    default void assert401ForMissingAuthToken(MockMvc mockMvc,
+                                              MockHttpServletRequestBuilder happyPathRequestBuilder,
+                                              String serviceName) throws Exception {
+
+        // GIVEN
+        HttpHeaders headers = createHttpHeaders(serviceName);
+        headers.remove(HttpHeaders.AUTHORIZATION); // <-- MISSING AUTH
+
+        // WHEN
+        mockMvc.perform(happyPathRequestBuilder.headers(headers))
+            // THEN
+            .andExpect(status().isUnauthorized());
+    }
+
+    /*
+     * CPO-33: “Implement tests for invalid S2S and Idam token”
+     * AC3: Mandatory parameters missing from the request (IDAM token invalid - MALFORMED)
+     */
+    default void assert401ForMalformedAuthToken(MockMvc mockMvc,
+                                                MockHttpServletRequestBuilder happyPathRequestBuilder,
+                                                String serviceName) throws Exception {
+
+        // GIVEN
+        HttpHeaders headers = createHttpHeaders(serviceName);
+        headers.remove(HttpHeaders.AUTHORIZATION);
+        headers.add(HttpHeaders.AUTHORIZATION, "MALFORMED"); // <-- MALFORMED AUTH
+
+        // WHEN
+        mockMvc.perform(happyPathRequestBuilder.headers(headers))
+            // THEN
+            .andExpect(status().isUnauthorized());
+    }
+
+    /*
+     * CPO-33: “Implement tests for invalid S2S and Idam token”
+     * AC3: Mandatory parameters missing from the request (IDAM token invalid - EXPIRED)
+     */
+    default void assert401ForExpiredAuthToken(MockMvc mockMvc,
+                                              MockHttpServletRequestBuilder happyPathRequestBuilder,
+                                              String serviceName) throws Exception {
+
+        // GIVEN
+        long authTtlMillis = -1;
+        HttpHeaders headers = createHttpHeaders(authTtlMillis, // <-- EXPIRED AUTH
+                                                serviceName,
+                                                AUTH_TOKEN_TTL);
+
+        // WHEN
+        mockMvc.perform(happyPathRequestBuilder.headers(headers))
+            // THEN
+            .andExpect(status().isUnauthorized());
```

Comment:
Can we assert response header `WWW-Authenticate` when there is an issue with oauth2 bear token?